addons/account/static/src/components/x2many_buttons/x2many_buttons.js

```diff
@@ -0,0 +1,58 @@
+/** @odoo-module **/
+
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+import { Component } from "@odoo/owl";
+
+class X2ManyButtons extends Component {
+    static template = "account.X2ManyButtons";
+    static props = standardFieldProps;
+
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+    }
+
+    async openTreeAndDiscard() {
+        const ids = this.currentField.currentIds;
+        await this.props.record.discard();
+        this.action.doAction({
+            type: "ir.actions.act_window",
+            res_model: this.currentField.resModel,
+            views: [
+                [false, "list"],
+                [false, "form"],
+            ],
+            domain: [["id", "in", ids]],
+        });
+    }
+
+    async openFormAndDiscard(id) {
+        await this.props.record.discard();
+        this.action.doAction({
+            type: "ir.actions.act_window",
+            res_model: this.currentField.resModel,
+            res_id: id,
+            views: [[false, "form"]],
+        });
+    }
+
+    get recordsCount() {
+        return this.currentField.count;
+    }
+
+    get loopRecordsCount() {
+        return this.currentField.count > 3 ? 4 : this.currentField.count;
+    }
```

Comment:
Are these really necessary?
I can probably justify `recordsCount` but not `loopRecordsCount` (and the latter could use the former - but better to remove it)